main/emacs/APKBUILD

```diff
@@ -0,0 +1,186 @@
+# Maintainer: stef <l0ls0fo2i@ctrlc.hu>
+# Contributor: Timo Teräs <timo.teras@iki.fi>
+
+pkgname=emacs
+pkgver=25.1
+pkgrel=0
+pkgdesc="The extensible, customizable, self-documenting real-time display editor"
+arch="noarch"
+url="http://www.gnu.org/software/emacs/emacs.html"
+license="GPL3"
+depends="hicolor-icon-theme desktop-file-utils"
+makedepends="autoconf automake linux-headers paxmark
+	librsvg-dev giflib-dev libxpm-dev gtk+2.0-dev gconf-dev alsa-lib-dev
+	imagemagick-dev glib-dev fontconfig-dev libpng-dev
+	libxml2-dev pango-dev tiff-dev libjpeg-turbo-dev ncurses-dev
+	ncurses-libs gnutls-dev libxaw-dev"
+subpackages="$pkgname-doc $pkgname-nox $pkgname-x11 $pkgname-gtk2"
+source="ftp://ftp.gnu.org/gnu/emacs/emacs-$pkgver.tar.xz
+        gamedir.patch
+        noaslr-dump.patch
+        musl.patch"
+
+_builddir="$srcdir/emacs-$pkgver"
+prepare() {
+	local i
+	cd "$_builddir"
+	for i in $source; do
+		case $i in
+		*.patch) msg $i; patch -p1 -i "$srcdir"/$i || return 1;;
+		esac
+	done
+}
+
+build_x11() {
+	cd "$_builddir/x11"
+	./autogen.sh || return 1
+	CFLAGS=-fno-pie \
+	LDFLAGS=-no-pie \
+	./configure \
+		--build=$CBUILD \
+		--host=$CHOST \
+		--prefix=/usr \
+		--sysconfdir=/etc \
+		--libexecdir=/usr/lib \
+		--localstatedir=/var \
+		--without-makeinfo \
+		--with-gameuser=:games \
+      --with-x-toolkit=athena \
+      --without-toolkit-scroll-bars \
+		--without-dbus \
+		--with-xft \
+		--with-jpeg=yes \
+		--with-tiff=yes \
+		|| return 1
+	make
+}
+
+build_gtk2() {
+	cd "$_builddir/gtk2"
+	./autogen.sh || return 1
+	CFLAGS=-fno-pie \
+	LDFLAGS=-no-pie \
+	./configure \
+		--build=$CBUILD \
+		--host=$CHOST \
+		--prefix=/usr \
+		--sysconfdir=/etc \
+		--libexecdir=/usr/lib \
+		--localstatedir=/var \
+		--without-makeinfo \
+		--with-gameuser=:games \
+		--with-x-toolkit=gtk2 \
+		--with-xft \
+		--with-jpeg=yes \
+		--with-tiff=no \
+		|| return 1
+	make
+}
+
+build_nox() {
+	cd "$_builddir/nox"
+	./autogen.sh || return 1
+	CFLAGS=-fno-pie \
+	LDFLAGS=-no-pie \
+	./configure \
+		--build=$CBUILD \
+		--host=$CHOST \
+		--prefix=/usr \
+		--sysconfdir=/etc \
+		--libexecdir=/usr/lib \
+		--localstatedir=/var \
+		--without-makeinfo \
+		--with-gameuser=:games \
+		--without-sound \
+		--without-x \
+		--without-file-notification \
+		|| return 1
+	make
+}
+
+build() {
+	cd "$_builddir"
+	[[ -d nox ]] || {
+      mkdir -p nox
+	   mv * nox || true
+	   cp -a nox gtk2
+	   cp -a nox x11
+   }
+   build_nox
+   build_x11
+   build_gtk2
+}
+
+package() {
+   mkdir -p "$pkgdir" && return 0
```

Comment:
This is nonsense. `&& return 0` is invoked only when `mkdir -p` returns 0 and since this is the last statement in the function, shell implicitly returns exit code of the last statement, i.e. this is like writing `true || true`.